Signal/src/call/CallService.swift

```diff
@@ -862,22 +876,19 @@ private class SignalCallData: NSObject {
         }
     }
 
-    private func handleIceDisconnected() {
+    private func handleIceDisconnected(callData: SignalCallData) {
         SwiftAssertIsOnMainThread(#function)
 
-        guard let call = self.call else {
-            // This will only be called for the current peerConnectionClient, so
-            // fail the current call.
-            OWSProdError(OWSAnalyticsEvents.callServiceCallMissing(), file: #file, function: #function, line: #line)
-            handleFailedCurrentCall(error: CallError.assertionError(description: "\(self.logTag) ignoring \(#function) since there is no current call."))
-            return
-        }
+        let call = callData.call
 
         Logger.info("\(self.logTag) in \(#function): \(call.identifiersForLogs).")
 
         switch call.state {
         case .remoteRinging:
             Logger.debug("\(self.logTag) in \(#function) disconnect while ringing... we'll keep ringing")
+        case .localRinging:
+            // This can happen if the incoming caller hangs up while we are ringing.
+            handleFailedCurrentCall(error: CallError.disconnected)
```

Comment:
This is a notable change.

Reply:
@charlesmchen-signal - could you respond to my comment here?

Reply:
Makes sense. This change isn't necessary now that you've added `localRinging` to the case above.